### PR TITLE
132 documentation users to members migration

### LINKED ISSUE #132: Docs update related to XbK API upgrade 26.0.1 and member migration
- new command "MigrateMembers"
- new setting "MemberIncludeUserSystemFields"

## Prerequisites & Compatibility

- target instance requirements raised to XbK API version 26.0.1

## New supported command for members migration

command moniker `--members`,
full example with other commands `migrate --siteId 1 --contact-management --media-libraries --custom-modules --settings-keys --page-types --pages --forms --attachments --culture en-US --sites --users --members`

in order to run members migration with custom fields and legacy system fields command `custom-modules` must be run before.

- apart from users, migration toolkit now supports migration of new entity `Member`
- member is migrated from users without access to administration (CMS_User with `UserPrivilegeLevelEnum.None`)
- migration to member supports:
  - corresponding users fields in member entity
  - creation and migration of no longer present fields optionally (configured with setting `Settings.MemberIncludeUserSystemFields`) from User and UserSettings
  - migration of custom fields from entities User and UserSettings
- migration to member NOT supports:
  - migration of passwords

## New setting

- `appsettings.json`, path `Settings.MemberIncludeUserSystemFields` for complete system fields migration (remove field You don't want to migrate): FirstName|MiddleName|LastName|FullName|PreferredCultureCode|PreferredUICultureCode|UserPrivilegeLevel|UserIsExternal|UserPasswordFormat|LastLogon|UserStartingAliasPath|UserLastModified|UserLastLogonInfo|UserIsHidden|UserIsDomain|UserHasAllowedCultures|UserMFRequired|UserMFSecret|UserMFTimestep|UserNickName|UserSignature|UserURLReferrer|UserCampaign|UserCustomData|UserRegistrationInfo|UserActivationDate|UserActivatedByUserID|UserTimeZoneID|UserAvatarID|UserGender|UserDateOfBirth|UserSettingsUserGUID|UserSettingsUserID|UserWaitingForApproval|UserDialogsConfiguration|UserDescription|UserAuthenticationGUID|UserSkype|UserIM|UserPhone|UserPosition|UserLogActivities|UserPasswordRequestHash|UserInvalidLogOnAttempts|UserInvalidLogOnAttemptsHash|UserPasswordLastChanged|UserAccountLockReason|UserShowIntroductionTile|UserDashboardApplications|UserDismissedSmartTips`

### PR DESCRIPTION
**Pull request by skurekjakub**
### Motivation

Which issue does this fix? 

Fixes #132

Added documentation includes
- supported package version update to 26.0.1
- instructions related to live site user migration to xbk members

FYI: my local VS code autofixed some markdown formatting, polluting the changeset slightly. I marked relevant added stuff with comments.

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [X] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,72 +32,79 @@ The Migration toolkit does not transfer all data available in the Kentico Xperie
...
+* **Media libraries and media files**
+  * Media library permissions are currently not supported in Xperience by Kentico, so are not migrated.
+* **Forms**
+  * The migration does not include the content of form autoresponder and notification emails. You can migrate form autoresponders to Xperience by Kentico manually by copying your HTML code and content into Email templates and Emails. See [Emails](https://docs.xperience.io/x/IaDWCQ).
+* **Users**
+
+  * Xperience by Kentico uses separate entities for users with access to the administration interface (_CMS_User_ table) and live site visitor accounts (_CMS_Member_ table). Consequently, only users whose _Privilege level_ is set to _Editor_ and above are migrated (_Users_ -> edit a user -> _General_ tab).
+  * Users in Xperience by Kentico must have an email address. Migration is only supported for users who have a unique email address value on the source instance.
+  * Custom user fields can be migrated together with _modules classes_.
+  * Live site users are represented using a separate **Member** entity and stored in the _CMS_Member_ table.
```
Comment (skurekjakub) [inline README.md:63]: Mentioned "members" as live site users under "Users"

File: README.md [hunk 2]
```diff
@@ -5,26 +5,25 @@
...
+* The source instance's database and file system must be accessible from the environment where you run the Migration toolkit.
 
 ### Target
 
-  * The toolkit is periodically updated to support migration to the **latest version** of Xperience by Kentico. However, there may be time gaps between Xperience by Kentico releases and Migration toolkit updates.
-	* Currently, Xperience by Kentico **23.0.3** is tested and supported.
-  * The target instance's database and file system must be accessible from the environment where you run the Migration toolkit.
-  * To avoid conflicts and inconsistencies, the target instance must not contain any data apart from an empty site and/or data from the source site created by previous runs of the Migration toolkit.
+* The toolkit is periodically updated to support migration to the **latest version** of Xperience by Kentico. However, there may be time gaps between Xperience by Kentico releases and Migration toolkit updates.
+  * Currently, Xperience by Kentico **26.0.1** is tested and supported.
```
Comment (skurekjakub) [inline README.md:24]: Updated the supported package version

File: Migration.Toolkit.CLI/README.md [hunk 3]
```diff
@@ -220,16 +275,18 @@ Add the options under the `Settings` section in the configuration file.
...
 | MigrationProtocolPath                                    | The absolute file system path of the location where the [migration protocol file](./MIGRATION_PROTOCOL_REFERENCE.md) is generated.<br /><br />For example: `"C:\\Logs\\Migration.Toolkit.Protocol.log"`               |
 | MigrateOnlyMediaFileInfo                                 | If set to `true`, only the database representations of media files are migrated, without the files in the media folder in the project's file system. For example, enable this option if your media library files are mapped to a shared directory or Cloud storage.<br /><br />If `false`, media files are migrated based on the `KxCmsDirPath` location.  |
+| MemberIncludeUserSystemFields | Determines which system fields from the *CMS_User* and *CMS_UserSettings* tables are migrated to *CMS_Member* in Xperience by Kentico. Fields that do not exist in *CMS_Member* are automatically created. <br /><br />The sample `appsettings.json` file included with the toolkit by default includes all user fields that can be migrated from Kentico Xperience 13. You exclude specific fields from the migration, by removing them from the setting."
```
Comment (skurekjakub) [inline Migration.Toolkit.CLI/README.md:278]: New setting key

File: Migration.Toolkit.CLI/README.md [hunk 4]
```diff
@@ -34,37 +34,38 @@ To perform the migration:
...
 | `--sites`                   | Enables migration of the [site](https://docs.xperience.io/x/34HFC). The site's basic properties are transferred to the target instance. Requires the `siteId` parameter to be specified. |  |
 | `--custom-modules`          | Enables migration of custom modules, [custom module classes and their data](https://docs.xperience.io/x/AKDWCQ), and [custom fields in supported system classes](https://docs.xperience.io/x/V6rWCQ).<br /><br />See: [Migration details for specific object types - Custom modules and classes](#custom-modules-and-classes)  | `--sites` |
 | `--users`                   | Enables migration of [users](https://docs.xperience.io/x/8ILWCQ) and [roles](https://docs.xperience.io/x/7IVwCg).<br /><br />See: [Migration details for specific object types - Users](#users) | `--sites`, `--custom-modules` |
+| `--members`                   | Enables migration of live site user accounts to [members](https://docs.xperience.io/x/BIsuCw). <br /><br />See: [Migration details for specific object types - Members](#members) | `--sites`, `--custom-modules` |
```
Comment (skurekjakub) [inline Migration.Toolkit.CLI/README.md:49]: New "members" CLI switch

**Comment by DavidBKentico** [comment]
Reivew notes: 

General - fix spacing in bullet lists (caused by the VS code autofix)

**Migration.Toolkit.CLI/Readme.md**
 * Migration details for specific object types > Members
   * "MemberIncludeUserSystemFields setting" > "MemberIncludeUserSystemFields setting configuration option" ? That's the terminology I used for this in other locations
    * "--custom-modues" typo
* Configuration
   * broken configuration option table after 'MemberIncludeUserSystemFields' ! (malformed markdown?)
  * MemberIncludeUserSystemFields - "You exclude specific fields from the migration, by removing them from the setting." > "Exclude specific fields from the migration by removing them from this configuration option."
  * I think it would be good to add MemberIncludeUserSystemFields to the Example code

**Comment by skurekjakub** [comment]
@DavidBKentico Fixed all review issues (you can check the individual commits)

**Comment by DavidBKentico** [comment]
Thanks, everything looks good now. Closing (I hope).

**Review by DavidBKentico — APPROVED** [review]

**Pull request merged**